.github/workflows/test.yaml

```diff
@@ -8,7 +8,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: [3.6, 3.7, 3.8, 3.9]
```

Comment:
On the top of this file there's a "on: [push]".

Can you replace by this: https://github.com/encode/starlette/blob/bc61505faef15b673bf4bf65db4927daa354d6b8/.github/workflows/test-suite.yml#L4 ?

Reply:
I'm personally strongly against adding this, it would have prevented me testing the CI on my fork and is an extra impediment to contributing.

https://github.com/hugovk/python-multipart/actions

Unless I edited the yml in my fork, and then remembered to revert it before creating a PR.
Or I can submit a PR on my branch first to check it passes -> extra impediment.

Or I can develop on my `master` -> not a best practice, we should use feature branches, and I only have a single `master` -> extra impediment.

The fact it's tested on upstream and on the fork isn't such a big deal: the testing is split between two different accounts and therefore doesn't use up any extra quota/time from any single account.

CIs allow us to test on a much larger matrix of Python versions and operating systems, much more than is possible locally and sometimes faster too. I sometimes avoid contributing to projects with no/broken CI. 

I believe the use of CIs are of huge benefit for software quality and development, especially when so easy nowadays; Travis CI really revolutionised this, remember how hard testing was before. We should rather encourage people to test on their own forks _before_ creating PRs.

Fail fast: shorten the time to finding a failure, enable quicker loops of incremental development.

Reply:
You're not against this. You're against GitHub blocking first time contributors for running their pipeline.

There's a setting to allow old GitHub users (3 months old ?) to not be blocked.

Reply:
> You're not against this. You're against GitHub blocking first time contributors for running their pipeline.

I don't mind that bit too much :)

---

Consider this workflow:

I come along and fork the repo and enable the CI on my fork.

I create a feature branch `add-3.10`.

I push my feature branch to my fork and check the CI.

I want the CI to test my changes on all supported Python versions (and possibly on several operating systems).

If it fails, I'll iterate: fix, push, repeat.

(This actually happened, see the first few failures as I iterated: https://github.com/hugovk/python-multipart/actions)

Once it passes, I can create a PR here, knowing that when the CI is enabled for the PR (we're 10 months and counting here), I'm confident it will pass and there's no surprises because I didn't test Python 3.7 or so on. 

However, if we have these branch restrictions:

```yaml
on:
  push:
    branches: ["master"]
  pull_request:
    branches: ["master"]
```

The CI won't build for me.

My options:

* Don't test on the CI, hope for the best when the PR runs here
* Edit the file and remove the restrictions, then before submitting the PR, remember to re-add the restrictions
* Develop in my `master` branch - not recommended, you can only work on one thing

All this applies even if I'm a second-time contributor.